packages/typespec-ts/test/integration/modelInheritanceEnumDiscriminator.spec.ts

```diff
@@ -0,0 +1,126 @@
+      const result = await client
+        .path("/type/model/inheritance/enum-discriminator/extensible-enum")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, validBody);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put extensible enum", async () => {
+    try {
+      const result = await client
+        .path("/type/model/inheritance/enum-discriminator/extensible-enum")
+        .put({ body: validBody });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get extensible enum if missing discriminator", async () => {
+    try {
+      const result = await client
+        .path(
+          "/type/model/inheritance/enum-discriminator/extensible-enum/missingdiscriminator"
+        )
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, { weight: 10 } as any);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get extensible enum if wrong discriminator", async () => {
+    try {
+      const result = await client
+        .path(
+          "/type/model/inheritance/enum-discriminator/extensible-enum/wrongdiscriminator"
+        )
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, { weight: 8, kind: "wrongKind" });
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get fixed enum", async () => {
+    try {
+      const result = await client
+        .path("/type/model/inheritance/enum-discriminator/fixed-enum")
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, validFixedEnumBody);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put fixed enum", async () => {
+    try {
+      const result = await client
+        .path("/type/model/inheritance/enum-discriminator/fixed-enum")
+        .put({ body: validFixedEnumBody });
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get fixed enum if missing discriminator", async () => {
+    try {
+      const result = await client
+        .path(
+          "/type/model/inheritance/enum-discriminator/fixed-enum/missingdiscriminator"
+        )
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, { length: 10 } as any);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get fixed enum if wrong discriminator", async () => {
+    try {
+      const result = await client
+        .path(
+          "/type/model/inheritance/enum-discriminator/fixed-enum/wrongdiscriminator"
+        )
+        .get();
+      assert.strictEqual(result.status, "200");
+      assert.deepEqual(result.body, { length: 8, kind: "wrongKind" });
```

Comment:
This case can pass because it's a inheritance and the response type is the sub class. 
```typescript
export interface SnakeOutputParent {
  /** Length of the snake */
  length: number;
  kind: string;
}

/** Cobra model */
export interface CobraOutput extends SnakeOutputParent {
  /** discriminator property */
  kind: "cobra";
}
```